### PR TITLE
Adding redisDB metadata

### PR DESCRIPTION
**Pull request by shivamkm07**
Signed-off-by: shivam <shivamkm07@gmail.com>

# Description

Adds `redisDB` metadata to redis Configuration Store 
## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #2409 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by berndverst** [comment]
Is this potentially a breaking change? Can you add a test to make sure that is not?

**Comment by shivamkm07** [comment]
> Is this potentially a breaking change? Can you add a test to make sure that is not? Specifically it should not impact pubsub and state store.

@berndverst It is not a breaking change as it just adds another metadata to config store. I have added a test case to verify that the default values are same as earlier, verifying it's not a breaking change.

It won't impact pubsub and statestore, as change is only in configuration store.

File: configuration/redis/redis.go [hunk 1]
```diff
@@ -269,19 +277,20 @@ func (r *ConfigurationStore) Subscribe(ctx context.Context, req *configuration.S
...
+		allKeysChannel := internal.GetRedisChannelFromKey("*", r.metadata.DB)
+		go r.doSubscribe(ctx, req, handler, allKeysChannel, subscribeID, stop)
 		return subscribeID, nil
 	}
 	for _, k := range req.Keys {
 		// subscribe single key
 		stop := make(chan struct{})
-		keySpacePrefixAndKey := keySpacePrefix + k
-		if oldStopChan, ok := r.subscribeStopChanMap.Load(keySpacePrefixAndKey); ok {
+		redisChannel := internal.GetRedisChannelFromKey(k, r.metadata.DB)
```
Comment (pravinpushkar) [inline configuration/redis/redis.go:287]: nit: `keyChannel`

**Review by pravinpushkar — APPROVED** [review]
lgtm

File: configuration/redis/internal/redis_value.go [hunk 2]
```diff
@@ -19,8 +19,8 @@ import (
 )
 
 const (
-	channelPrefix = "__keyspace@0__:"
-	separator     = "||"
+	keySpacePrefix = "__keyspace@"
```
[the remark this answers is not in the document]
Comment (shivamkm07) [reply configuration/redis/internal/redis_value.go:22]: In redis for subscribing to any key update, there is a specific channel that needs to be subscribed: `__keyspace@<DB>__:<KEY>`. So basically if DB is 0(default value), the channel Prefix to KEY would be `__keyspace@0__:` which was what set here earlier. 

I this PR, I am just trying to make the DB configurable, as users might want to store different set of configurations in different DBs. It is already in place for redis state store currently: https://docs.dapr.io/reference/components-reference/supported-state-stores/setup-redis/#component-format and there was an ask for this as well sometime back, but the issue got closed due to no activity: https://github.com/dapr/components-contrib/issues/2000

File: configuration/redis/redis.go [hunk 3]
```diff
@@ -131,6 +130,15 @@ func parseRedisMetadata(meta configuration.Metadata) (metadata, error) {
 		}
 	}
 
+	m.DB = defaultDB
+	if val, ok := meta.Properties[redisDB]; ok && val != "" {
+		parsedVal, err := strconv.ParseInt(val, defaultBase, defaultBitSize)
+		if err != nil {
+			return m, fmt.Errorf("redis store error: can't parse redisDB field: %s", err)
```
Comment (pravinpushkar) [inline configuration/redis/redis.go:137]:
```suggestion
			return m, fmt.Errorf("redis store error: can't parse redisDB field from metadata: %w", err)
```

**Review by berndverst — APPROVED** [review]

**Comment by berndverst** [comment]
Please make sure to also cover this scenario in the Redis configuration certification test.

**Pull request merged**